File: PHP/Hardware.php
<?php 

    // Hardware merupakan kelas orang tua kedua dengan urutan
    // Produk <- Hardware <- Memory

    class Hardware extends Product
    {
        // atribut publik    
        public $brand;
        public $model;

        // Konstruktor
        public function __construct($brand = "", $model = "")
        {
            $this->brand = $brand;
            $this->model = $model;            
        }   

        // metode untuk menyimpan data brand
        public function setBrand($brand)
        {
            $this->brand = $brand;
        }   
        public function getBrand()
        {
            return $this->brand;
        }

        // metode untuk menyimpan data model
        public function setModel($model)
        {
            $this->model = $model;
        }   
        public function getModel()
        {
            return $this->model;
        }

        // Destruktor untuk membersihkan memory
        public function __destruct(){
            
        }
    }
?>

File: Cpp/Hardware.cpp
// Hardware merupakan kelas orang tua kedua dengan urutan
// Produk <- Hardware <- Memory

class Hardware : public Product
{
	// atribut private
	private:
		string brand;
		string model;

	// atribut publik	
	public:
		// parameter konstruktor
		Hardware(string brand, string model)
		{
			this->brand = brand;
			this->model = model;
		}
		// Konstruktor
		Hardware(){
			this->brand = "";
			this->model = "";
		}

		// untuk menyimpan data brand
		void setBrand(string brand)
		{
			this->brand = brand;
		}

		string getBrand()
		{
			return this->brand;
		}

		// untuk menyimpan data model
		void setModel(string model)
		{
			this->model = model;
		}

		string getModel()
		{
			return this->model;
		}

		// Destruktor untuk membersihkan memory
		~Hardware(){

		}

};